### ISSUE TITLE
403 forbidden

### ISSUE THREAD

**Issue by dgleba** [osteel/docker-tutorial#3]
Hi, I am following your blog post.
[http://tech.osteel.me/posts/2015/12/18/from-vagrant-to-docker-how-to-use-docker-for-local-web-development.html](url)

I got the first step - welcome to nginx.

on the second step, i get..

```
403 Forbidden
nginx/1.11.3
```

Can you help?

thanks in advance!

**Comment by homerakshay** [comment]
@dgleba Hello, are you having a go-template of any kind for the redirection. Also are you using any templates for the same. Only then nginx will know what to redirect on what port. 

Regards,
Akshay

**Comment by osteel** [comment]
Can you _ls -al_ the content of _/var/www/html_ of your Nginx container and paste the result here?

```
$ docker ps
$ docker exec -it [CONTAINER ID] /bin/bash
$ cd /var/www/html
$ ls -al
```

**Comment by dgleba** [comment]
The index.php is not showing up in the container.

I see others have similar issues.

Similar issue.. but this one is win10..
https://github.com/docker/docker/issues/22981

# 

![289f-volumehostfilenotshowingincontainer](https://cloud.githubusercontent.com/assets/4622715/18598025/dacb3128-7c1e-11e6-9f29-0c8c0cced79b.png)

**Comment by dgleba** [comment]
Thank you for your fantastic tutorial. I am learning a lot.

I have run the same docker definition on three hosts, Win7/8, ubuntu 14.04, and OSX. All three get different results.

Windows cannot see the index.php, but phpmyadmin works.
Linux can see index.php, but phpmyadmin is not accessible.
See screen shots.

Files I ran are here: https://github.com/dgleba/dock289e2.git

### Edit:

It works on OSX. I go to  http://localhost and see the doctor saying hello, and on http://localhost:8078 I get phpmyadmin running.

Windows:
![mingw64__c_n_dropbox_csd_serve_dkr_dock289e2](https://cloud.githubusercontent.com/assets/4622715/18608900/6566ec4c-7cc3-11e6-8aa6-78d72fa6f4dc.png)

Linux:
![vamp206b ubuntu14 04 -oracle vm virtualbox](https://cloud.githubusercontent.com/assets/4622715/18608903/6d893a9c-7cc3-11e6-966d-5587dcde80ba.png)

**Comment by osteel** [comment]
Hmm being on macos myself, I cannot offer much support I'm afraid.

What you can do however, is try to mount your host directory into the _nginx_ and _php_ containers directly, without using the data-only container _app_, and see how that goes:

```
nginx:
    build: ./nginx/
    ports:
        - 80:80
    links:
        - php
    volumes:
        - .:/var/www/html

php:
    build: ./php/
    expose:
        - 9000
    links:
        - mysql
    volumes:
        - .:/var/www/html
```

**Comment by d3haynes** [comment]
if I ls-al the content of /var/www/html on nginx container
![dock](https://cloud.githubusercontent.com/assets/2005366/18918820/faf44a3a-8569-11e6-825b-9568cb2431e5.jpg)

**Comment by osteel** [comment]
@d3haynes have you tried what I suggested above?

**Comment by d3haynes** [comment]
Hi @osteel ,

I did, and this is my result. 

![osteel-2](https://cloud.githubusercontent.com/assets/2005366/18953207/47483246-861b-11e6-8256-1e259028d7de.jpg)

Thank you for your guidance, any help is appreciated.

**Comment by osteel** [comment]
Hmm then again, not being on Windows myself I can't even try to replicate this.

Are you using Docker Toolbox or the native Windows app?

**Comment by d3haynes** [comment]
Docker toolbox unfortunately...I had windows 7 pro but when it updated to 10 it put me on home edition, so I cannot use Hyper V :(

**Comment by osteel** [comment]
Bummer!

I'm afraid I can't offer you any more help on this, sorry :-/

**Comment by d3haynes** [comment]
I am now trying to just use a host vm running linux, and run docker on that, still running into issues, so we will see what happens. thanks for the help!

**Comment by tominon** [comment]
You need to check C drive in the Docker Settings as a shared drive first and then the directory contents should be visible inside the guest:

![image](https://cloud.githubusercontent.com/assets/508519/19613362/52eca798-97ec-11e6-8b8e-143f58e7a512.png)

**Comment by osteel** [comment]
Closing in absence of any follow-up.

**Comment by grok** [comment]
Here's a potential solution:

I am actually not using this repo -- but I ran into the same problem as the other person.
I got mine working. I'm using an identical nginx/default.conf -- nothing fancy.

```
FROM nginx:latest

COPY ./default.conf /etc/nginx/conf.d/default.conf
```

```
mariadb:
  image: mariadb:latest
  volumes_from:
    - mariadb_data
  environment:
    MYSQL_ROOT_PASSWORD: secret
    MYSQL_DATABASE: project
    MYSQL_USER: project
    MYSQL_PASSWORD: project
  expose:
    - 3306

mariadb_data:
  image: tianon/true
  volumes:
    - /var/lib/mariadb

nginx:
  build: ./.build/docker/nginx/
  ports:
      - 80:80
  links:
    - php
  volumes:
    - ./web:/var/www/html

php:
  build: ./.build/docker/php/
  expose:
    - 9000
  links:
    - mariadb
  volumes:
    - ./web:/var/www/html
```

Here's the fix: Don't try to hit localhost when the directory is empty.

Here's the error I get:

> 2017/02/09 05:46:48 [error] 7#7: *2 directory index of "/var/www/html/" is forbidden, client: 172.17.0.1, server: , request: "GET / HTTP/1.1", host: "localhost"

Once I added index.php -- drop in a phpinfo():
PHP Version 7.0.14

All gravy.

I hope this helps the previous guy and future people who might stumble on this result.

**Comment by Paganus** [comment]
I got the problem (using linux), and the problem was that I was using VM (according Docker official manual https://docs.docker.com/get-started/part4/). So volumes points at ./www/html on the VM, and obviously it doesn't exist.
But if I tried to run docker-compose up -d it was saying "Couldn't connect to Docker daemon - you might need to run `docker-machine start default`". That's why I thought I must use VM. Then I tried docker-compose up with sudo (without starting any VM) and everything works nice, except it needs sudo every time now.

P.S. Thank you for the tutorial!
